From 6159fb90563d15cce2a46359606aa1cea9f75a02 Mon Sep 17 00:00:00 2001
From: Karthik Kambatla <kasha@cloudera.com>
Date: Thu, 9 May 2013 15:38:10 -0700
Subject: [PATCH 059/596] MR1: CLOUDERA-BUILD. CDH-9235. Fix API
 incompatibility in
 mapreduce.lib.jobcontrol.JobControl (cherry picked
 from commit
 5ab1a1731b97dcdc4877960cb955ae74a9bbaa24)

(cherry picked from commit 1a3f6b32b1ab742bf4ffa5e8a644dafcee835b8f)
(cherry picked from commit cc4cab6dddc8d1437d859b7dc31c81ff6d0d0e17)
(cherry picked from commit b0d1022fd4eca614d9640db430230b6b8ed85ba7)
---
 .../mapreduce/lib/jobcontrol/JobControl.java       |   11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapreduce/lib/jobcontrol/JobControl.java b/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapreduce/lib/jobcontrol/JobControl.java
index 4ec384f..aaf598c 100644
--- a/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapreduce/lib/jobcontrol/JobControl.java
+++ b/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapreduce/lib/jobcontrol/JobControl.java
@@ -22,6 +22,7 @@
 import java.util.Collection;
 import java.util.Hashtable;
 import java.util.Map;
+import java.util.List;
 
 import org.apache.hadoop.mapreduce.lib.jobcontrol.ControlledJob.State;
 
@@ -87,32 +88,32 @@ public JobControl(String groupName) {
   /**
    * @return the jobs in the waiting state
    */
-  public ArrayList<ControlledJob> getWaitingJobList() {
+  public List<ControlledJob> getWaitingJobList() {
     return toArrayList(this.waitingJobs);
   }
 	
   /**
    * @return the jobs in the running state
    */
-  public ArrayList<ControlledJob> getRunningJobList() {
+  public List<ControlledJob> getRunningJobList() {
     return toArrayList(this.runningJobs);
   }
 	
   /**
    * @return the jobs in the ready state
    */
-  public ArrayList<ControlledJob> getReadyJobsList() {
+  public List<ControlledJob> getReadyJobsList() {
     return toArrayList(this.readyJobs);
   }
 	
   /**
    * @return the jobs in the success state
    */
-  public ArrayList<ControlledJob> getSuccessfulJobList() {
+  public List<ControlledJob> getSuccessfulJobList() {
     return toArrayList(this.successfulJobs);
   }
 	
-  public ArrayList<ControlledJob> getFailedJobList() {
+  public List<ControlledJob> getFailedJobList() {
     return toArrayList(this.failedJobs);
   }
 	
-- 
1.7.9.5

